### PR TITLE
Bind the widget UI to the current suggestions

### LINKED ISSUE #5402: Bind the UI to the current suggestions in SharedStorage
When the widget is refreshed, read the contents from ExtensionKit.SharedStorage and update the suggestions list.

Depends on #5401

### PR DESCRIPTION
**Pull request by dshr**
## What's this?
The connection between the UI and the suggestions data. Also connecting a button to open the app.

### Relationships
Closes #5402

## Why do we want this?
We want the UI to work, don't we?

## How is it done?
By connecting the appropriate DataSource and populating the cells.

### Why not another way?
Seems simple enough 🤷‍♂ 

## :squid: Permissions
🎛 crew only.

### CODE REVIEW

File: Toggl.iOS.ExtensionKit/SharedStorage.Suggestions.cs [hunk 1]
```diff
@@ -28,6 +28,8 @@ public void SetCurrentSuggestions(IList<Suggestion> suggestions)
         public IList<Suggestion> GetCurrentSuggestions()
         {
             var array = userDefaults.ArrayForKey(suggestionsKey);
+            if (array == null)
+                return null;
             return array.Cast<NSDictionary>().Select(fromDictionary).ToList();
```
Comment (heytherewill) [inline Toggl.iOS.ExtensionKit/SharedStorage.Suggestions.cs]: Why not simply:

```suggestion
            return array?.Cast<NSDictionary>().Select(fromDictionary).ToList();
```

File: Toggl.iOS.TimerWidgetExtension/SuggestionTableViewCell.cs [hunk 2]
```diff
@@ -1,12 +1,39 @@
...
+            DescriptionLabel.Text = suggestion.Description;
+            if (suggestion.ProjectName != string.Empty)
+            {
+                ProjectLabel.Text = suggestion.ProjectName;
+                DotView.BackgroundColor = new Color(suggestion.ProjectColor).ToNativeColor();
+                ProjectLabel.Hidden = DotView.Hidden = false;
+            }
+            else
+            {
+                ProjectLabel.Hidden = DotView.Hidden = true;
```
Comment (heytherewill) [inline Toggl.iOS.TimerWidgetExtension/SuggestionTableViewCell.cs]: Double assignments are really hard to understand when scanning code; I'd rather see this as two lines

File: Toggl.iOS.TimerWidgetExtension/SuggestionsDataSource.cs [hunk 3]
```diff
@@ -1,22 +1,42 @@
...
 
 namespace Toggl.iOS.TimerWidgetExtension
 {
-    public class SuggestionsDataSource : UITableViewDataSource
+    public class SuggestionsDataSource : UITableViewSource
     {
         private const string identifier = "SuggestionCell";
 
+        private IList<Suggestion> suggestions;
+        public IList<Suggestion> Suggestions
```
Comment (heytherewill) [inline Toggl.iOS.TimerWidgetExtension/SuggestionsDataSource.cs]: No need for a backing field; simply make the getter private

File: Toggl.iOS.TimerWidgetExtension/SuggestionsDataSource.cs [hunk 4]
```diff
@@ -1,22 +1,42 @@
...
         {
             var cell = tableView.DequeueReusableCell(identifier, indexPath) as SuggestionTableViewCell;
+            var suggestion = suggestions[indexPath.Row];
+            cell.PopulateCell(suggestion);
             return cell;
         }
 
-        public override nint RowsInSection(UITableView tableView, nint section)
+        public override nint RowsInSection(UITableView tableview, nint section)
+        {
```
Comment (heytherewill) [inline Toggl.iOS.TimerWidgetExtension/SuggestionsDataSource.cs]: => suggestions.Count;

File: Toggl.iOS.ExtensionKit/SharedStorage.Suggestions.cs (hunk 1 above)
Comment (dshr) [reply Toggl.iOS.ExtensionKit/SharedStorage.Suggestions.cs]: Didn't know we could do this 🙈

File: Toggl.iOS.TimerWidgetExtension/SuggestionsDataSource.cs [hunk 5]
```diff
@@ -1,22 +1,32 @@
...
         {
             var cell = tableView.DequeueReusableCell(identifier, indexPath) as SuggestionTableViewCell;
+            var suggestion = Suggestions[indexPath.Row];
+            cell.PopulateCell(suggestion);
             return cell;
         }
 
-        public override nint RowsInSection(UITableView tableView, nint section)
+        public override nint RowsInSection(UITableView tableview, nint section)
+            => return Suggestions.Count;
```
[the remark this answers is not in the document]
Comment (dshr) [reply Toggl.iOS.TimerWidgetExtension/SuggestionsDataSource.cs]: Oh, got too quick lol

**Pull request merged**